src/framework/components/joint/component.js

```diff
@@ -0,0 +1,371 @@
+        return this._entityB;
+    }
+
+    set breakForce(force) {
+        if (this._constraint && this._breakForce !== force) {
+            this._constraint.setBreakingImpulseThreshold(force);
+            this._breakForce = force;
+        }
+    }
+
+    get breakForce() {
+        return this._breakForce;
+    }
+
+    set enableCollision(enableCollision) {
+        this._destroyConstraint();
+        this._enableCollision = enableCollision;
+        this._createConstraint();
+    }
+
+    get enableCollision() {
+        return this._enableCollision;
+    }
+
+    set angularLimitsX(limits) {
+        if (!this._angularLimitsX.equals(limits)) {
+            this._angularLimitsX.copy(limits);
+            this._updateAngularLimits();
+        }
+    }
+
+    get angularLimitsX() {
+        return this._angularLimitsX;
+    }
+
+    set angularLimitsY(limits) {
+        if (!this._angularLimitsY.equals(limits)) {
+            this._angularLimitsY.copy(limits);
+            this._updateAngularLimits();
+        }
+    }
+
+    get angularLimitsY() {
+        return this._angularLimitsY;
+    }
+
+    set angularLimitsZ(limits) {
+        if (!this._angularLimitsZ.equals(limits)) {
+            this._angularLimitsZ.copy(limits);
+            this._updateAngularLimits();
+        }
+    }
+
+    get angularLimitsZ() {
+        return this._angularLimitsZ;
+    }
+
+    set linearLimitsX(limits) {
+        if (!this._linearLimitsX.equals(limits)) {
+            this._linearLimitsX.copy(limits);
+            this._updateLinearLimits();
+        }
+    }
+
+    get linearLimitsX() {
+        return this._linearLimitsX;
+    }
+
+    set linearLimitsY(limits) {
+        if (!this._linearLimitsY.equals(limits)) {
+            this._linearLimitsY.copy(limits);
+            this._updateLinearLimits();
+        }
+    }
+
+    get linearLimitsY() {
+        return this._linearLimitsY;
+    }
+
+    set linearLimitsZ(limits) {
+        if (!this._linearLimitsZ.equals(limits)) {
+            this._linearLimitsZ.copy(limits);
+            this._updateLinearLimits();
+        }
+    }
+
+    get linearLimitsZ() {
+        return this._linearLimitsZ;
+    }
+
+    _convertTransform(pcTransform, ammoTransform) {
+        const pos = pcTransform.getTranslation();
+        const rot = new Quat();
+        rot.setFromMat4(pcTransform);
+
+        const ammoVec = new Ammo.btVector3(pos.x, pos.y, pos.z);
+        const ammoQuat = new Ammo.btQuaternion(rot.x, rot.y, rot.z, rot.w);
+
+        ammoTransform.setOrigin(ammoVec);
+        ammoTransform.setRotation(ammoQuat);
+
+        Ammo.destroy(ammoVec);
+        Ammo.destroy(ammoQuat);
+    }
+
+    _updateAngularLimits() {
+        const constraint = this._constraint;
+        if (constraint) {
+            const alx = this._angularLimitsX;
+            const aly = this._angularLimitsY;
+            const alz = this._angularLimitsZ;
+
+            const limits = new Ammo.btVector3(alx.x, aly.x, alz.x);
```

Comment:
It's surprisingly not easy to do that actually. Because Ammo may or may not be loaded. >99% of PlayCanvas apps will not even use the joint component, so doing any static allocations is perhaps more to be avoided here.